src/libraries/System.Private.CoreLib/src/System/Collections/Hashtable.cs

```diff
@@ -415,6 +416,17 @@ private uint InitHash(object key, int hashsize, out uint seed, out uint incr)
         //
         public virtual void Add(object key, object? value)
         {
+#if TARGET_BROWSER
+            if (_keycomparer == null)
```

Comment:
We need to move this logic to constructor, because there are other methods which are able to modify the collection. like `public virtual object? this[object key]`

Reply:
Also please link the issue which explains motivations.

Reply:
How can we move it to constructor when we don't know what type will be added to `Hashtable`? If we add `int` then it will throw because we set comparer to `StringComparer`.

Reply:
could somebody add `"1"` string and then `2` int ? That would fail anyway, right ?

Reply:
No, the problem is that for `Hashtable` it's perfectly allowed, even when not recommended.

Reply:
Even if we assume that users don't mix types, constructor does not know what types will be put into the non-generic collection.